### PR TITLE
Support mongo object literals in agg.pipeline

### LINKED ISSUE #2771: QScript MongoDB planner resorts to map-reduce when creating new object literals
The MongoDB planner appears to resort to map-reduce any time a new object literal needs to be created as part of a plan, this is unnecessary and a regression from the legacy MongoDB planner.

To illustrate, consider the query
```sql
select { "foo" : pop } as bar from zips
```
Legacy Planner (MongoDB 3.2)
```
db.zips.aggregate(
  [{ "$project": { "bar": { "foo": "$pop" } } }],
  { "allowDiskUse": true });
```
QScript Planner (MongoDB 3.2)
```
db.zips.mapReduce(
  function () {
    emit.apply(
      null,
      (function (key, value) { return [key, { "bar": { "foo": value.pop } }] })(
        this._id,
        this))
  },
  function (key, values) { return values[0] },
  { "out": { "replace": "tmp.gen_0", "db": "testdata" } });
db.tmp.gen_0.aggregate(
  [{ "$project": { "bar": "$value.bar" } }],
  { "allowDiskUse": true });
```

### PR DESCRIPTION
**Pull request by rintcius**
Fixes #2771

### CODE REVIEW

File: mongodb/src/test/scala/quasar/physical/mongodb/planner.scala [hunk 1]
```diff
@@ -479,6 +479,15 @@ class PlannerSpec extends
              ListMap())))
     }
 
+    "plan select map" in {
```
Comment (rintcius) [inline mongodb/src/test/scala/quasar/physical/mongodb/planner.scala]: the example from #2771 does not use `objectLit` but added the test anyway (& not hitting map-reduce anymore)

**Comment by rintcius** [comment]
After the many planner test changes for `arrayLit`s, this 1 test change is a bit disappointing. That may change though after #2681 gets addressed.

File: mongodb/src/test/scala/quasar/physical/mongodb/planner.scala (hunk 1 above)
Comment (wemrysi) [reply mongodb/src/test/scala/quasar/physical/mongodb/planner.scala]: I think a minor revision to `select {"p": pop}, state from zips` should involve `objectLit`.

File: mongodb/src/main/scala/quasar/physical/mongodb/planner/StaticHandler.scala [hunk 2]
```diff
@@ -42,8 +45,18 @@ object StaticHandler {
...
+
+      def toBsonFieldName(ej: T[EJson]): Option[BsonField.Name] =
+        (CommonEJson.prj(ej.project) >>= (str.getOption(_))).map(BsonField.Name(_))
+
       def handle[A](fm: FreeMapA[T, A]): Option[EX[FreeMapA[T, A]]] =
-        None
+        fm.project match {
+          case MapFuncCore.StaticMap(m) =>
+            val x: Option[List[(BsonField.Name, FreeMapA[T, A])]] =
+              m.map(t => toBsonFieldName(t._1) map ((_, t._2))).sequence
```
Comment (wemrysi) [inline mongodb/src/main/scala/quasar/physical/mongodb/planner/StaticHandler.scala]: :golf: `m.traverse(t => ...)`

File: mongodb/src/test/scala/quasar/physical/mongodb/planner.scala (hunk 1 above)
Comment (rintcius) [reply mongodb/src/test/scala/quasar/physical/mongodb/planner.scala]: 👍  added that test

**Pull request merged**